akka-stream/src/main/scala/akka/stream/scaladsl/Compression.scala

```diff
@@ -16,10 +18,17 @@ object Compression {
    * coming out of the flow can be fully decompressed without waiting for additional data. This may
    * come at a compression performance cost for very small chunks.
    *
-   * FIXME: should compression level / strategy / flush mode be configurable? See https://github.com/akka/akka/issues/21849
+   * FIXME: should strategy / flush mode be configurable? See https://github.com/akka/akka/issues/21849
+   */
+  def gzip: Flow[ByteString, ByteString, NotUsed] = gzip()
+
+  /**
+   * Same as [[gzip]] with a custom level.
+   *
+   * @param level Compression level (0-9)
    */
-  def gzip: Flow[ByteString, ByteString, NotUsed] =
-    CompressionUtils.compressorFlow(() ⇒ new GzipCompressor)
+  def gzip(level: Int = Deflater.BEST_COMPRESSION): Flow[ByteString, ByteString, NotUsed] =
```

Comment:
Wait a moment, don't use default parameters, this will be hard to binary compatibly evolve - won't it?

Reply:
Esp since https://github.com/akka/akka/pull/23845/files#diff-1ab43228652167c43fedb322ba567895R48 may need a parameter